### PR TITLE
added take_along_axis

### LINKED ISSUE #11549: take_along_axis

### PR DESCRIPTION
**Pull request by rajshukla1102**
Close #11549

### CODE REVIEW

**Comment by rajshukla1102** [comment]
@vaatsalya123 can you please review this pr😊

**Comment by AnnaTz** [comment]
Hi there, `jax.numpy.take_along_axis` also has a `mode` argument. I have just added support for this feature to `ivy.take_along_axis`. Could you please modify your function and test accordingly? Thanks!

**Comment by rajshukla1102** [comment]
@AnnaTz yeah sure!

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_indexing.py [hunk 1]
```diff
@@ -133,3 +133,41 @@ def test_jax_numpy_diag_indices(
...
+@handle_frontend_test(
+    fn_tree="jax.numpy.take_along_axis",
+    dtype_x_indices_axis=helpers.array_indices_axis(
+        array_dtypes=helpers.get_dtypes("numeric"),
+        indices_dtypes=["int32", "int64"],
+        min_num_dims=1,
+        max_num_dims=5,
+        min_dim_size=1,
+        max_dim_size=10,
+        indices_same_dims=True,
```
Comment (AnnaTz) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_indexing.py:148]: The `mode` argument defines how out-of-bounds indices are handled. To properly test this we need to pass `valid_bounds=False` to `array_indices_axis` as is done in `test_take_along_axis`.

**Review by AnnaTz — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_indexing.py (hunk 1 above)
Comment (rajshukla1102) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_indexing.py:148]: @AnnaTz backend implementation for numpy jax torch tf still talks 3 pos arguments! can i implement the changes over there also?

Comment (AnnaTz) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_indexing.py:148]: Could you please link the functions you're referring to?

Comment (AnnaTz) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_indexing.py:148]: This is because `np.take_along_axis` does not natively support `mode`. Our implementation of `take_along_axis` adds support by wrapping the original function in the backend. You should not make changes to the backend functions.

Comment (rajshukla1102) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_jax_numpy_indexing.py:148]: yeah I just checked `numpy.take_along_axis` implementation! Thankyou!

**Comment by rajshukla1102** [comment]
@AnnaTz can review this?

File: ivy/functional/frontends/jax/numpy/indexing.py [hunk 2]
```diff
@@ -19,3 +19,9 @@ def diag(v, k=0):
 def diag_indices(n, ndim=2):
     idx = ivy.arange(n, dtype=int)
     return (idx,) * ndim
+
+
+# take_along_axis
+@to_ivy_arrays_and_back
+def take_along_axis(arr, indices, axis, mode='fill'):
+    return ivy.take_along_axis(arr, indices, axis, mode)
```
Comment (AnnaTz) [inline ivy/functional/frontends/jax/numpy/indexing.py:27]: mode should be passed as a keyword argument

**Review by AnnaTz — CHANGES_REQUESTED** [review]

File: ivy/functional/frontends/jax/numpy/indexing.py (hunk 2 above)
Comment (AnnaTz) [reply ivy/functional/frontends/jax/numpy/indexing.py:27]: `def take_along_axis(arr, indices, axis, mode='fill'):` was correct. But the next line should be `return ivy.take_along_axis(arr, indices, axis, mode=mode)` because `mode` is a keyword argument.

**Review by AnnaTz — CHANGES_REQUESTED** [review]

**Comment by rajshukla1102** [comment]
do we need to change [here](https://github.com/unifyai/ivy/blob/a066eb34e4e8f71d455a4d62aa7f9a33c8df56fb/ivy/functional/frontends/numpy/indexing_routines/indexing_like_operations.py#L25) also?

**Comment by rajshukla1102** [comment]
@AnnaTz is this okay?

**Comment by AnnaTz** [comment]
Great! Ready to be merged 🙂

**Comment by rajshukla1102** [comment]
hey @AnnaTz thankyou!😊
